indico/migrations/versions/20190118_1514_416f9c877300_create_reservationlink_model.py

```diff
@@ -0,0 +1,88 @@
+"""Create ReservationLink model
+
+Revision ID: 416f9c877300
+Revises: cbe630695800
+Create Date: 2019-01-18 15:14:29.749357
+"""
+
+import sqlalchemy as sa
+from alembic import op
+
+from indico.core.db.sqlalchemy import PyIntEnum
+from indico.core.db.sqlalchemy.links import LinkType
+
+
+# revision identifiers, used by Alembic.
+revision = '416f9c877300'
+down_revision = 'ff49d8d05ce7'
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    op.create_table(
+        'reservation_links',
+        sa.Column('id', sa.Integer(), nullable=False),
+        sa.Column('event_id', sa.Integer(), nullable=True, index=True),
+        sa.Column('linked_event_id', sa.Integer(), nullable=True, index=True),
+        sa.Column('contribution_id', sa.Integer(), nullable=True, index=True),
+        sa.Column('session_block_id', sa.Integer(), nullable=True, index=True),
+        sa.Column('link_type',
+                  PyIntEnum(LinkType, exclude_values={LinkType.category, LinkType.subcontribution, LinkType.session}),
+                  nullable=False),
+        sa.ForeignKeyConstraint(['contribution_id'], ['events.contributions.id']),
+        sa.ForeignKeyConstraint(['event_id'], ['events.events.id']),
+        sa.ForeignKeyConstraint(['linked_event_id'], ['events.events.id']),
+        sa.ForeignKeyConstraint(['session_block_id'], ['events.session_blocks.id']),
+        sa.PrimaryKeyConstraint('id'),
+        schema='roombooking'
+    )
+
+    op.add_column('reservations', sa.Column('link_id', sa.Integer(), nullable=True), schema='roombooking')
+    op.create_index(None, 'reservations', ['link_id'], unique=False, schema='roombooking')
+    op.create_foreign_key(None, 'reservations', 'reservation_links', ['link_id'], ['id'], source_schema='roombooking',
+                          referent_schema='roombooking')
+
+    # Migrate reservations.event_id to new reservation_links and set
+    # reservations.link_id based on the id of the newly created row.
+    op.execute('''
+        WITH reserv_data AS (
+            SELECT nextval('roombooking.reservation_links_id_seq') AS new_link_id, id AS reserv_id, event_id
+            FROM roombooking.reservations
+            WHERE event_id IS NOT NULL
+        ), link_ids_data AS (
+            INSERT INTO roombooking.reservation_links (id, event_id, linked_event_id, link_type)
+            SELECT new_link_id, event_id, event_id, 2 -- LinkType.event
+            FROM reserv_data
+            RETURNING id AS link_id
+        )
+        UPDATE roombooking.reservations
+        SET link_id = link_ids_data.link_id
+        FROM link_ids_data,reserv_data
+        WHERE id = reserv_data.reserv_id and link_ids_data.link_id = reserv_data.new_link_id;
+    ''')
+
+    op.drop_column('reservations', 'event_id', schema='roombooking')
+
+
+def downgrade():
+    op.add_column('reservations', sa.Column('event_id', sa.Integer(), autoincrement=False, nullable=True),
```

Comment:
`autoincrement=False` can be omitted, alembic is just overzealous when autogenerating this